### DISCUSSION TITLE
Warnings present in bason gem

### DISCUSSION THREAD

**Discussion by nitinsingh25** [mongodb/bson-ruby#304] [General]
Currently Bason gem produces following warnings, we must fix it.

<pre>
lib/bson/decimal128/builder.rb:192: warning: assigned but unused variable - original
lib/bson/decimal128/builder.rb:193: warning: assigned but unused variable - e
lib/bson/decimal128/builder.rb:194: warning: assigned but unused variable - decimal
lib/bson/decimal128/builder.rb:320: warning: assigned but unused variable - base
lib/bson_native.bundle: warning: method redefined; discarding old next_object_id
lib/bson/object_id.rb:371: warning: previous definition of next_object_id was here</pre>

**Comment by alexbevi** [comment]
Hi @nitinsingh25, can you share a reproduction that produces these warnings as well as what version of the `bson` library is being used?

**Reply by nitinsingh25** [reply]
My Ruby version is 2.7.7 & I tried it with latest bson gem 4.15.0 as well. There is no specific step for that, actually I am using mongo  gem and when I run my tests it produces following warning.
These warnings are pretty clear & we can see those clearly in code that variable is assigned but not used anywhere.

**Comment by alexbevi** [comment]
Thanks for the report @nitinsingh25. I've opened RUBY-3228 to investigate this further.
